updater/dart_bindings/lib/src/bindings.dart

```diff
@@ -38,7 +38,10 @@ class AppParameters extends ffi.Struct {
     if (updateUrl != null) {
       config.ref.update_url = updateUrl.toNativeUtf8();
     }
-    config.ref.original_libapp_path = libappPath.toNativeUtf8();
+    config.ref.original_libapp_paths = calloc<ffi.Pointer<Utf8>>();
```

Comment:
This seems wrong?  It looks like it's allocating a pointer to utf8.  It needs to allocate an array of pointers (at a fixed size), no?

Reply:
I think this is the only real problem in this patch. I think this code is writing out of bounds and we're just not catching it because we're not using libgmalloc or similar.

Reply:
I think if you passed the size + 1 it would "just work".
https://api.dart.dev/stable/2.19.4/dart-ffi/Allocator/allocate.html

Reply:
I think I addressed this in the latest commit, lmk what you think.